### ISSUE TITLE
rc --follow-location finds no symbol

### ISSUE THREAD

**Issue by BinaryKhaos** [Andersbakken/rtags#797]
I tried rtags in Emacs today but ran into the problem that rtags-find-symbol-at-point does not work. Using rtags-find-symbol though works just fine. After further debugging, "rc --follow-location" never finds anything.

I tried to debug this quite some time now but gdb is having a rough time with the data structures and I suppose someone close to the project can more easily spot what is going on, so for now I settle for a simple bug report. :-)

My rtags checkout is from today (92a7c74b73d8ab6a83a256f526f278583c52722f) and I am using rtags.el from MELPA (rtags-20160915.901).

If there is anything I can do to help, please let me know.

**Comment by Andersbakken** [comment]
Hi Matthias

That's a little odd. There's a chance that there's some incompatibility
between the version of the elisp version in melpa and your rtags checkout
though it doesn't seem likely.

Can you try the following little self-test: (It'll take a few minutes)

<script>
git clone git@github.com:Andersbakken/rtags.git --recursive
cd rtags
cmake . -DCMAKE_EXPORT_COMPILE_COMMANDS=1
make
./bin/rdm &
sleep 2 && rc -J .
while [ "`rc --is-indexing`" =  "1" ]; do
    sleep 1
done
./bin/rc --follow-location src/ClassHierarchyJob.cpp:66:5:
./bin/rc -q
</script>

When it's done it should print out something like this:

src/ClassHierarchyJob.cpp:40:76:            std::function<void (const
Symbol &, const char *title, int, FindFunc)> recurse = [&](const Symbol
&sym,

Can you see if this works?

My suspicion is that something is going wrong with the elisp.

Anders

**Comment by BinaryKhaos** [comment]
Thanks Anders for your invested time and help.

I am afraid, your tests (with a few fixes :P) worked fine. I also previously debugged the elisp function to see if something was going wrong there, which it didn't. So I decided (stupidly without thinking) to kill the rtags cache folder and try it again on my project (with my initial rtags checkout). And lo and behold, it worked fine now on my project. Since I deleted the cache, I can no longer check what was wrong. I tried recreating the bug and did many tests and crazy things but it worked just like expected.

So... I guess I can close the issue as I can no longer reproduce it. Should I ever run into this again, I will make sure to _not_ delete the cache folder though. ;-)

Sorry for the noise, and thanks again.
